sale_check_preparation_time/models/stock.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2020 Akretion France (http://www.akretion.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import api, fields, models
+import pytz
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    @api.multi
+    def action_done(self):
+        res = super(StockPicking, self).action_done()
+        for picking in filter(lambda x: x.picking_type_id.code == 'outgoing' and
+                              x.sale_id, self):
+            if picking.company_id.tz:
+                local_tz = pytz.timezone(picking.company_id.tz)
+            else:
+                local_tz = pytz.timezone('UTC')
```

Comment:
Maybe it's better to add a flag "check_preparation_time" on company and then put in mandatory the field
- tz and order_limit_hour.
So here the field will be always fill?
And if the check_preparation_time is not tic then we just skip this step?

Reply:
@sebastienbeau , Indeed, if you wish to use the functionality, the field tz and order_limit_hour will be always fill.